File: arrays/rev_str.py
def reverse (string):
    if not isinstance(string, str):
        raise AttributeError("not a string")
    else:
        #method 1
        newstring = string[::-1]

        #method 2
        #newstring = ''.join(reversed(string))

        #method 3
        #newstring = ''
        #for char in string:
            #newstring = char + reversed_string

        print(newstring)


reverse("hello there")

File: arrays/rev_str.js
function reverse(str) {
    if(!str || typeof str != 'string' || str.length < 2) return 'invalid input';

    const newstr = [];
    const strlen = str.length-1

    for(let i = strlen; i >= 0; i--) {
        newstr.push(str[i]);

    }

    return newstr.join('');
}

function reverse2(str) {
    if(!str || typeof str != 'string' || str.length < 2) return 'invalid input';
    
    return str.split('').reverse().join('');
}

const reverse3 = str => [...str].reverse().join('');

console.log(reverse2('hello there'));